browser/ui/views/tabs/brave_tab_drag_context.cc

```diff
@@ -0,0 +1,297 @@
+  return original_context_->IsDragSessionActive();
+}
+
+bool BraveTabDragContext::IsActiveDropTarget() const {
+  return original_context_->IsActiveDropTarget();
+}
+
+std::vector<int> BraveTabDragContext::GetTabXCoordinates() const {
+  return original_context_->GetTabXCoordinates();
+}
+
+int BraveTabDragContext::BraveTabDragContext::GetActiveTabWidth() const {
+  return original_context_->GetActiveTabWidth();
+}
+
+int BraveTabDragContext::BraveTabDragContext::GetTabDragAreaWidth() const {
+  return original_context_->GetTabDragAreaWidth();
+}
+
+int BraveTabDragContext::BraveTabDragContext::TabDragAreaEndX() const {
+  return original_context_->TabDragAreaEndX();
+}
+
+int BraveTabDragContext::BraveTabDragContext::TabDragAreaBeginX() const {
+  return original_context_->TabDragAreaBeginX();
+}
+
+int BraveTabDragContext::BraveTabDragContext::GetHorizontalDragThreshold()
+    const {
+  return original_context_->GetHorizontalDragThreshold();
+}
+
+void BraveTabDragContext::SetBoundsForDrag(
+    const std::vector<TabSlotView*>& views,
+    const std::vector<gfx::Rect>& bounds) {
+  return original_context_->SetBoundsForDrag(views, bounds);
+}
+
+void BraveTabDragContext::StartedDragging(
+    const std::vector<TabSlotView*>& views) {
+  return original_context_->StartedDragging(views);
+}
+
+void BraveTabDragContext::DraggedTabsDetached() {
+  original_context_->DraggedTabsDetached();
+}
+
+void BraveTabDragContext::StoppedDragging(
+    const std::vector<TabSlotView*>& views,
+    const std::vector<int>& initial_positions,
+    bool completed) {
+  original_context_->StoppedDragging(views, initial_positions, completed);
+}
+
+void BraveTabDragContext::LayoutDraggedViewsAt(
+    const std::vector<TabSlotView*>& views,
+    TabSlotView* source_view,
+    const gfx::Point& location,
+    bool initial_drag) {
+  original_context_->LayoutDraggedViewsAt(views, source_view, location,
+                                          initial_drag);
+}
+
+void BraveTabDragContext::ForceLayout() {
+  original_context_->ForceLayout();
+}
+
+int BraveTabDragContext::GetInsertionIndexForDraggedBounds(
+    const gfx::Rect& dragged_bounds,
+    std::vector<TabSlotView*> dragged_views,
+    int num_dragged_tabs,
+    absl::optional<tab_groups::TabGroupId> group) const {
+  if (!tabs::ShouldShowVerticalTabs()) {
+    return original_context_->GetInsertionIndexForDraggedBounds(
+        dragged_bounds, dragged_views, num_dragged_tabs, group);
+  }
+
+  // If the strip has no tabs, the only position to insert at is 0.
+  if (!GetTabCount())
+    return 0;
+
+  absl::optional<int> index;
+  // If we're dragging a group by its header, the first element of
+  // |dragged_views| is a group header, and the second one is the first tab
+  // in that group.
+  int first_dragged_tab_index = group.has_value() ? 1 : 0;
+  if (static_cast<size_t>(first_dragged_tab_index) >= dragged_views.size()) {
+    // TODO(tbergquist): This shouldn't happen, but we're getting crashes
```

Comment:
what is the crash stack trace? Can you replicate?

Reply:
This code is copied from upstream? That's a major problem that needs to be resolved. What is different between BraveTabDragContext and TabDragContextImpl? We absolutely cannot have copied code here because it will get out of date and potentially be a problem even when vertical tabs are not enabled.